### PR TITLE
I888 fixing tooltip icon error

### LINKED ISSUE #888: The tooltips of the Warning icon next to the uploaded image are incorrect after trying to upload incompatible files
[Affected Platforms]:
- Windows 10 x64
- Ubuntu 16.04 x64
- macOS 10.14

[Affected Versions]:
- Firefox Nightly 75.0a1 (2020-02-19) Build ID 20200219095352
- Firefox Beta 74.0b5 , Build ID 20200218224219 
- Firefox Release 73.0.1 Build ID 20200207195153

[Prerequisites]:
- Have the latest version of the Firefox Color add-on modified in order to link to the stage website.
- Have the previous Firefox Color add-on installed through "about:debugging".

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. Navigate to the https://color.stage.mozaws.net/ website.
3. Navigate to "Custom backgrounds" tab and click the "...Or add your own" button.
4. Upload a picture up to 1MB (jpg, png or bmp).
5. Click the "Replace image" button.
6. Upload a file bigger than 1MB.
7. Hover the Warning icon from the left side of the image's thumbnail.
8. Click the "Replace image" button.
9. Upload a file that is not an image (eg. xpi, zip, pdf, js etc).
10. Hover the Warning icon from the left side of the image's thumbnail.

[Expected result]:
- Step 7 - The Warning tooltip contains the same text as the error message "The image is too large (1MG maximum size)" OR nothing happens.
- Step 10 - The Warning tooltip contains the same text as the error message "The file is not an accepted image type" OR nothing happens.

[Actual result]:
- Step 7 - {"too large":true, "wrongType":false} tooltip is displayed.
- Step 10 - {"too large":false, "wrongType":true} tooltip is displayed.

[Notes]:
- I have attached a screen recording of the issue:

color

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #888

### CODE REVIEW

File: src/web/lib/components/ThemeCustomBackgroundPicker/index.js [hunk 1]
```diff
@@ -129,12 +139,7 @@ export class ThemeCustomBackgroundPicker extends React.Component {
                   <React.Fragment>
                     <Modal>
                       <ul className="errors">
-                        {errors.tooLarge && (
-                          <li>The image is too large. (1MB maximum size)</li>
-                        )}
-                        {errors.wrongType && (
-                          <li>The file is not an accepted image type.</li>
-                        )}
+                        <li>{mapErrors(errors)}</li>
```
Comment (Rob--W) [inline src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: Previously, more than one error message could be shown.

With this change, only one message is shown.

Comment (rebmullin) [reply src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: hmm i see..I'll revisit. thanks!

**Comment by rebmullin** [comment]
hey @Rob--W, how is it now?

File: src/web/lib/components/ThemeCustomBackgroundPicker/index.js [hunk 2]
```diff
@@ -19,6 +19,21 @@ import "./index.scss";
 import iconHAlignLeft from "./icon_align_left.svg";
 import iconVAlignCenter from "./icon_align_center.svg";
 
+
+const mapErrors = ({ tooLarge, wrongType}) => {
+  if (tooLarge && wrongType) {
+    return `
```
Comment (Rob--W) [inline src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: This function is used both for generating the HTML and title attribute. HTML is insensitive to line breaks (treated as whitespace), but the title attribute is not. I think that with this change, you may end up with a tooltip that has a leading and trailing blank line.

To account for this, how about returning an array here, and let the callers format the message as they see fit? For example:

HTML:

```
mapErrors(errors).map(msg => `<li>${msg}</li>`);
```

Title:
```
mapErrors(errors).join("\n")
```

Comment (rebmullin) [reply src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: yep :) i agree that's a lot better!

File: src/web/lib/components/ThemeCustomBackgroundPicker/index.js [hunk 3]
```diff
@@ -129,12 +141,7 @@ export class ThemeCustomBackgroundPicker extends React.Component {
                   <React.Fragment>
                     <Modal>
                       <ul className="errors">
-                        {errors.tooLarge && (
-                          <li>The image is too large. (1MB maximum size)</li>
-                        )}
-                        {errors.wrongType && (
-                          <li>The file is not an accepted image type.</li>
-                        )}
+                        {mapErrors(errors).map((error, i) => <li key={i}>{error}</li>)}
```
Comment (Rob--W) [inline src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: Is `key={i}` doing anything useful? If not, let's remove it for simplicity.

Comment (rebmullin) [reply src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: this is required by react. when mapping over items you need a key. do you prefer another name? or i can try using the error name..?

Comment (Rob--W) [reply src/web/lib/components/ThemeCustomBackgroundPicker/index.js]: I wasn't aware of the `key` requirement (https://reactjs.org/docs/lists-and-keys.html#keys), thanks for the explanation.

**Pull request merged**